src/libraries/System.Private.CoreLib/src/System/RuntimeType.cs

```diff
@@ -139,6 +139,143 @@ public override Array GetEnumValues()
             return ret;
         }
 
+        /// <summary>
+        /// Retrieves an array of the values of the underlying type constants in a specified enumeration type.
+        /// </summary>
+        /// <remarks>
+        /// This method can be used to get enumeration values when creating an array of the enumeration type is challenging.
+        /// For example, <see cref="T:System.Reflection.MetadataLoadContext" /> or on a platform where runtime codegen is not available.
+        /// </remarks>
+        /// <returns>An array that contains the values of the underlying type constants in enumType.</returns>
+        /// <exception cref="ArgumentException">
+        /// Thrown when the type is not Enum
+        /// </exception>
+        public override Array GetEnumValuesAsUnderlyingType()
+        {
+            if (!IsActualEnum)
+                throw new ArgumentException(SR.Arg_MustBeEnum, "enumType");
+
+            // Get all of the values
+            ulong[] values = Enum.InternalGetValues(this);
+
+#if MONO
+            switch (GetTypeCode(Enum.GetUnderlyingType(this)))
```

Comment:
```suggestion
            switch (RuntimeTypeHandle.GetCorElementType(Enum.InternalGetUnderlyingType(this)))
```
I think this should work and then you won't need the rest of the ifdefs.

Reply:
Or you can do a switch on `GetTypeCode(this)`. I think that will work well on both CoreCLR and Mono without anyifdefs.